examples/demo/modules/blog/cms_blog_api.e

```diff
@@ -8,93 +8,108 @@ class
 			else
-				create {CMS_BLOG_STORAGE_NULL} node_storage.make
+				create {CMS_BLOG_STORAGE_NULL} blog_storage.make
 			end
-			initialize_node_types
+--			initialize_node_types
 		end
 
+feature {CMS_API_ACCESS, CMS_MODULE, CMS_API} -- Restricted access		
+
+	node_api: CMS_NODE_API
+
 feature {CMS_MODULE} -- Access nodes storage.
 
-	node_storage: CMS_BLOG_STORAGE_I
+	blog_storage: CMS_BLOG_STORAGE_I
+
+feature -- Configuration of blog handlers
+
+	entries_per_page : NATURAL_32 = 2
+			-- The numbers of posts that are shown on one page. If there are more post a pagination is generated
+			--| For test reasons this is 2, so we don't have to create a lot of blog entries.
+			--| TODO: Set to bigger constant.
 
 feature -- Access node
 
 	blogs_count: INTEGER_64
-			-- Number of nodes of type blog
+			-- Number of nodes of type blog.
 		do
-			Result := node_storage.blogs_count
+			Result := blog_storage.blogs_count
 		end
 
-	blogs_count_from_user (user_id: INTEGER_64) : INTEGER_64
-			-- Number of nodes of type blog from user with user_id
+	blogs_count_from_user (a_user: CMS_USER): INTEGER_64
+			-- Number of nodes of type blog from user with `a_user_id'.
+		require
+			has_id: a_user.has_id
 		do
-			Result := node_storage.blogs_count_from_user(user_id)
+			Result := blog_storage.blogs_count_from_user (a_user)
 		end
 
-	blogs_order_created_desc: LIST[CMS_NODE]
+	blogs_order_created_desc: LIST [CMS_BLOG]
 			-- List of nodes ordered by creation date (descending)
 		do
-			Result := add_authors(node_storage.blogs)
+			Result := nodes_to_blogs (blog_storage.blogs)
 		end
 
-	blogs_order_created_desc_limited (a_limit:NATURAL_32; a_offset:NATURAL_32) : LIST[CMS_NODE]
+	blogs_order_created_desc_limited (a_limit: NATURAL_32; a_offset: NATURAL_32): LIST [CMS_BLOG]
 			-- List of nodes ordered by creation date and limited by limit and offset
-		local
-			tmp: LIST[CMS_NODE]
 		do
-			-- load all posts and add the authors to each post
-			Result := add_authors(node_storage.blogs_limited (a_limit, a_offset))
-
+				-- load all posts and add the authors to each post
+			Result := nodes_to_blogs (blog_storage.blogs_limited (a_limit, a_offset))
 		end
 
-	blogs_from_user_order_created_desc_limited (a_user_id: INTEGER_32; a_limit:NATURAL_32; a_offset:NATURAL_32) : LIST[CMS_NODE]
+	blogs_from_user_order_created_desc_limited (a_user: CMS_USER; a_limit: NATURAL_32; a_offset: NATURAL_32) : LIST [CMS_BLOG]
 			-- List of nodes ordered by creation date and limited by limit and offset
-		local
-			tmp: LIST[CMS_NODE]
+		require
+			has_id: a_user.has_id
 		do
-			-- load all posts and add the authors to each post
-			Result := add_authors(node_storage.blogs_from_user_limited (a_user_id, a_limit, a_offset))
-
+				-- load all posts and add the authors to each post
+			Result := nodes_to_blogs (blog_storage.blogs_from_user_limited (a_user, a_limit, a_offset))
 		end
 
 feature {NONE} -- Helpers
 
-	add_authors(posts: LIST[CMS_NODE]) : LIST[CMS_NODE]
```

Comment:
add_authors was a bad name for a function.
in Eiffel. we tend to use verb for procedure, and noun for function

you could have name it   "nodes_filled_with_author" for instance, but here, I prefer to use a nodes_to_blogs function.